src/udev/udev-builtin.c

```diff
@@ -102,18 +102,19 @@ enum udev_builtin_cmd udev_builtin_lookup(const char *command) {
 }
 
 int udev_builtin_run(struct udev_device *dev, enum udev_builtin_cmd cmd, const char *command, bool test) {
-        char arg[UTIL_PATH_SIZE];
-        int argc;
-        char *argv[128];
+        _cleanup_strv_free_ char **argv = NULL;
+        int r;
 
         if (!builtins[cmd])
                 return -EOPNOTSUPP;
 
+        r = strv_split_extract(&argv, command, NULL, EXTRACT_QUOTES|EXTRACT_RELAX|EXTRACT_CUNESCAPE|EXTRACT_CUNESCAPE_RELAX);
+        if (r < 0)
+                return r;
+
         /* we need '0' here to reset the internal state */
         optind = 0;
-        strscpy(arg, sizeof(arg), command);
-        udev_build_argv(arg, &argc, argv);
-        return builtins[cmd]->cmd(dev, argc, argv, test);
+        return builtins[cmd]->cmd(dev, strv_length(argv), argv, test);
```

Comment:
Does `strv_split_extract` implement the same semantics? Do we have some tests for this?